src/test/TestUtils.cpp

```diff
@@ -83,6 +83,16 @@ TestInvariantManager::handleInvariantFailure(
 TestApplication::TestApplication(VirtualClock& clock, Config const& cfg)
     : ApplicationImpl(clock, cfg)
 {
+    // ApplicationImpl has registered the signals and handler at this point.
```

Comment:
Sadly, there are many calls to `Application::create` in test code (they don't use `TestApplication`), so this would only work for some test cases - for example see "bucket persistence over app restart".

So either change that or find a different way

Reply:
Ahh that's unfortunate. What do you think about using a `static bool` in `Application` that `runTest` in `test.cpp` toggles on so we know that test cases are running? We can then check the flag in the signal handler.

Reply:
could also just use a special config "TESTS_ENABLED" bool instead of a static